### PR TITLE
Refactor app use connection details endpoint

### PR DESCRIPTION
**Pull request by Ocupe**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+LIVEKIT_API_SECRET=
+# URL pointing to the LiveKit server. (example: `wss://my-livekit-project.livekit.cloud`)
+LIVEKIT_URL=
 
-# URL pointing to the LiveKit server.
-LIVEKIT_URL=wss://my-livekit-project.livekit.cloud
 
+# OPTIONAL SETTINGS
+# ################# 
+# NEXT_PUBLIC_CONN_DETAILS_ENDPOINT=
```
Comment (lukasIO) [inline .env.example:15]: this should probably be under the `# publich` section?

File: .env.example [hunk 2]
```diff
@@ -16,9 +21,6 @@ LIVEKIT_URL=wss://my-livekit-project.livekit.cloud
 # S3_REGION=
 
 # PUBLIC
-NEXT_PUBLIC_LK_TOKEN_ENDPOINT=/api/token
-#NEXT_PUBLIC_LK_RECORD_ENDPOINT=/api/record
```
Comment (lukasIO) [inline .env.example:20]: why was the record endpoint removed entirely?

Comment (Ocupe) [reply .env.example:20]: Exidentially removed. Good catch!

File: .env.example [hunk 3]
```diff
@@ -16,11 +20,10 @@ LIVEKIT_URL=wss://my-livekit-project.livekit.cloud
...
-
 # Uncomment settings menu when using a LiveKit Cloud, it'll enable Krisp noise filters.
 # NEXT_PUBLIC_SHOW_SETTINGS_MENU=true
+# NEXT_PUBLIC_CONN_DETAILS_ENDPOINT=
```
Comment (lukasIO) [inline .env.example:25]: I think we should still fill at least this value in, as it won't change for users under normal circumstances

File: app/api/connection-details/route.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      return new NextResponse('Missing required query parameter: roomName', { status: 400 });
+    }
+    if (participantName === null) {
+      return new NextResponse('Missing required query parameter: participantName', { status: 400 });
+    }
+
+    // Generate participant token
+    const participantToken = await createParticipantToken(
+      {
+        identity: `${participantName}__${randomUUID()}`,
```
Comment (lukasIO) [inline app/api/connection-details/route.ts:27]: nice!

File: app/api/connection-details/route.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        name: participantName,
+        metadata,
+      },
+      roomName,
+    );
+
+    // Return connection details
+    const data: ConnectionDetails = {
+      server_url: LIVEKIT_URL!,
+      room_name: roomName ? roomName : 'playground_hello-world',
```
Comment (lukasIO) [inline app/api/connection-details/route.ts:37]: what's the incentive behind providing a fallback room? I think for meet we should expect the roomName to always be set?

File: app/api/connection-details/route.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      participant_token: participantToken,
+      participant_name: participantName ? participantName : 'playground-default-participant',
```
Comment (lukasIO) [inline app/api/connection-details/route.ts:39]: same as above, I think we should require a participantName for meet

**Review by lukasIO — APPROVED** [review]
couple more nits

File: .env.example (hunk 3 above)
Comment (Ocupe) [reply .env.example:25]: Good catch. I would rather remove it. We set it to a hardcoded default value in the code. It just adds noise to the .env file.

File: app/api/connection-details/route.ts (hunk 5 above)
Comment (Ocupe) [reply app/api/connection-details/route.ts:37]: Good catch! Hardcoded test stuff.

File: app/api/connection-details/route.ts (hunk 6 above)
Comment (Ocupe) [reply app/api/connection-details/route.ts:39]: Good catch! Hardcoded test stuff.

**Pull request merged**